clients/adapter/src/main/java/org/eclipse/hono/client/registry/DeviceRegistrationClient.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+
+package org.eclipse.hono.client.registry;
+
+import io.opentracing.SpanContext;
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * A client for accessing Hono's Device Registration API.
+ * <p>
+ * See Hono's <a href="https://www.eclipse.org/hono/docs/api/device-registration/">
+ * Device Registration API</a> for a description of the status codes returned.
+ */
+public interface DeviceRegistrationClient {
```

Comment:
I haven't spent much time on the interfaces for the registry clients. We haven't talked about whether we want registry clients with other transport protocols. But the way the interfaces look, there is nothing AMQP-specific in them anyway.

Reply:
Yes, that was the intention. So, in the future we might as well implement the interfaces based on HTTP ...